src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/AbstractCodeActionOrUserDiagnosticTest.cs

```diff
@@ -681,6 +683,46 @@ static void VerifyExpectedDocumentText(string expected, string actual)
         protected virtual ImmutableArray<CodeAction> MassageActions(ImmutableArray<CodeAction> actions)
             => actions;
 
+        internal static void VerifyCodeActionsRegisteredByProvider(List<CodeFix> fixes)
+        {
+            var diagnosticsAndEquivalenceKeyToTitleMap = new Dictionary<(Diagnostic diagnostic, string equivalenceKey), string>();
+            foreach (var fix in fixes)
+            {
+                var codeAction = fix.Action;
+                foreach (var diagnostic in fix.Diagnostics)
+                {
+                    var key = (diagnostic, codeAction.EquivalenceKey);
+                    var existingTitle = diagnosticsAndEquivalenceKeyToTitleMap.GetOrAdd(key, _ => codeAction.Title);
+                    if (existingTitle != codeAction.Title)
+                    {
+                        Assert.False(true, @$"Expected different 'CodeAction.EquivalenceKey' for code actions registered for same diagnostic:
```

Comment:
Verified that populate switch tests fail with this assert. I will push follow-up commit(s) in this PR to fix all fixers/refactorings where this fire.